src/rpg_database/character_sheet/LevelAndXPCrossDependencyCheck.java

```diff
@@ -0,0 +1,98 @@
+package rpg_database.character_sheet;
+
+import java.security.InvalidParameterException;
+import java.util.HashMap;
+
+import rpg_database.character_sheet.exceptions.CrossDependencyException;
+import rpg_database.character_sheet.interfaces.MultipleFieldsGetterSetter;
+
+public class LevelAndXPCrossDependencyCheck implements MultipleFieldsGetterSetter<LevelAndXPCrossDependencyCheck, Integer> {
+
+	private int xp;
+	private int level;
+	private final int maxLevel = 22;
+	private final int maxXp = 70000;
+
+	private final HashMap<Integer, Integer> xpNeedingForLevelMap = fillUpWithLevelXpPairs();
+
+	public LevelAndXPCrossDependencyCheck() {
+		xp = 0;
+		level = 1;
+	}
+
+	public int GetXP() {
+		return xp;
+	}
+
+	@Override
+	public Class<LevelAndXPCrossDependencyCheck> getImplementingClass() {
+		return LevelAndXPCrossDependencyCheck.class;
+	}
+
+	@Override
+	public Class<Integer> getDataTypeClass() {
+		return Integer.class;
+	}
+
+	@Override
+	public void setSelfValueByField(Fields field, Integer value) {
+		if (field == Fields.XP) {
+			if (value > maxXp)
+				throw new CrossDependencyException("The xp gap is: " + maxXp + "!");
+			if (level == 1 || xpNeedingForLevelMap.get(level) <= value) {
+				xp = value;
+				while (xpNeedingForLevelMap.get(level) < xp && xpNeedingForLevelMap.get(level + 1) <= xp) {
+					level++;
+				}
+			} else {
+				throw new CrossDependencyException("Your level is higher than the given xp!");
+			}
+		} else if (field == Fields.LEVEL) {
+			if (value > maxLevel)
+				throw new CrossDependencyException("The level gap is: " + maxLevel + "!");
+			if (xp == 0 || xpNeedingForLevelMap.get(value) <= xp) {
+				level = value;
+				xp = xpNeedingForLevelMap.get(value);
+			} else
+				throw new CrossDependencyException("You don't have enough xp for this level!");
+		} else {
+			throw new InvalidParameterException("It's not a valid Field type!");
+		}
+	}
+
+	@Override
+	public Integer getStoredValueByField(Fields field) {

```

Comment:
ezt feljebb is írtam, de iderakom mert szerintem fontos:
Amúgy ez a multiplefieldsgettersetter megközelítés nem rossz ötlet, viszont érdemes észben tartanod, hogy ez egy olyan interfész ami arra készült, hogy tulajdonképpen különböző fieldeket legyen képes (GOLD, SILVER, BRONZE) inputként fogadni és outputként visszaadni amelyek ugyanazon érték alapján számolódnak. Ezzel szemben, te két fieldnek két külön értéket tárolsz. Ha szerinted ez egy MultipleGetterSetter eset, akkor gondolkodj el azon, hogy vajon tényleg van -e értelme két értékben tárolnod ugyanazt az információt.